src/main/scala/com/mwz/sonar/scala/scapegoat/ScapegoatSensor.scala

```diff
@@ -0,0 +1,193 @@
+/*
+ * Sonar Scala Plugin
+ * Copyright (C) 2018 All contributors
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+package com.mwz.sonar.scala
+package scapegoat
+
+import inspections.ScapegoatInspection.AllScapegoatInspections
+
+import cats.implicits._
+import com.mwz.sonar.scala.util.JavaOptionals._
+import com.mwz.sonar.scala.util.Log
+import org.sonar.api.batch.fs.{FileSystem, InputFile}
+import org.sonar.api.batch.rule.ActiveRule
+import org.sonar.api.batch.sensor.{Sensor, SensorContext, SensorDescriptor}
+import org.sonar.api.batch.sensor.issue.NewIssue
+import org.sonar.api.config.Configuration
+import scalariform.ScalaVersion
+
+import java.nio.file.{Path, Paths}
+import scala.util.{Failure, Success, Try}
+
+/** Main sensor for importing Scapegoat reports to SonarQube */
+final class ScapegoatSensor extends ScapegoatSensorInternal with ScapegoatReportParser
+
+/** Implementation of the sensor */
+private[scapegoat] abstract class ScapegoatSensorInternal extends Sensor {
+  // cake pattern to mock the scapegoat report parser in tests
+  scapegoatReportParser: ScapegoatReportParserAPI =>
+
+  import ScapegoatSensorInternal._
```

Comment:
Although I generally agree that having all imports in the top of the file is the best.
I have my exceptions, and particularly with importing values from a companion object, I would expect to have that import inside the class instead of in the top of the file.

If you're concerned only by sonar, we can add a comment to disable the rule. But if you have any other reason for not having the import there, I will be happy to hear it.